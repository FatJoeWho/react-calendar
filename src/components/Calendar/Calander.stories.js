import React from "react";
import { storiesOf } from "@storybook/react";
import Calendar from "components/Calendar";
import Moment from "moment";

const _date = new Moment();

storiesOf("Calendar", module).add("Basic Month", () =>
	<Calendar month={_date} />
);
