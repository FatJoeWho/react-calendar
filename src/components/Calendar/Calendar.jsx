// @flow

import React from "react";
import Moment from "moment";
import CalendarDay from "components/CalendarDay";
import styled from "styled-components";

export default class Calendar extends React.Component {
	static defaultProps = {
		momentJsInstance: Moment()
	};

	nextMonth = e => {
		e.preventDefault();
		this.setState({
			momentJsInstance: this.props.momentJsInstance.add(1, "M")
		});
	};

	prevMonth = e => {
		e.preventDefault();
		this.setState({
			momentJsInstance: this.props.momentJsInstance.subtract(1, "M")
		});
	};

	getDayList = () => {
		return Array(this.props.momentJsInstance.daysInMonth())
			.fill()
			.map((_: mixed, dayNumber: number): Array<any> => {
				let dayInstance = this.props.momentJsInstance
					.clone() //clone so that it is disposed in memory garbage collection
					.date(dayNumber + 1); //0 index counting
				return (
					<CalendarDay
						key={dayNumber}
						fromOtherMonth={false}
						gridColumn={dayInstance.day() + 1}
						displayContent={dayInstance.format("D")}
					/>
				);
			});
	};

	render() {
		const dayList = this.getDayList();
		return (
			<div>
				<header>
					<button onClick={this.prevMonth}>&larr;</button>
					<div>
						{this.props.momentJsInstance.format("MMMM YYYY")}
					</div>
					<button onClick={this.nextMonth}>&rarr;</button>
				</header>
				<DayHeadersDiv>
					<div>Sun</div>
					<div>Mon</div>
					<div>Tues</div>
					<div>Wed</div>
					<div>Thurs</div>
					<div>Fri</div>
					<div>Sat</div>
				</DayHeadersDiv>
				<MonthDiv>
					{dayList}
				</MonthDiv>
			</div>
		);
	}
}

const DayHeadersDiv = styled.div`
		display: grid;
		grid-template-columns: repeat(7, 1fr);
		grid-gap: 10px;
		grid-auto-rows: minmax(1fr, auto);
		text-align: center;
	`,
	MonthDiv = styled.div`
		display: grid;
		grid-template-columns: repeat(7, 1fr);
		grid-gap: 10px;
		border: 1px solid black;
	`;
