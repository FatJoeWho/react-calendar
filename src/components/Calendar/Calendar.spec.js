import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Calendar from "components/Calendar";
import Moment from "moment";

const fakeEvent = {
	preventDefault: () => {
		return;
	}
};

describe("Calendar", () => {
	it("Should render", () => {
		const _date = new Moment("2 August 2017"),
			component = renderer.create(<Calendar month={_date} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should fetch an array of days", () => {
		const _date = new Moment("2 August 2017"),
			component = mount(<Calendar month={_date} />),
			days = component.instance().getDayList();
		expect(days.length).toEqual(31);
	});

	it("Should go to next and prev month", () => {
		const _date = new Moment("2 August 2017"),
			component = mount(<Calendar month={_date} />);

		expect(component.props().momentJsInstance.format("M")).toEqual("8");
		component.instance().nextMonth(fakeEvent);
		expect(component.props().momentJsInstance.format("M")).toEqual("9");
		component.instance().prevMonth(fakeEvent);
		component.instance().prevMonth(fakeEvent);
		expect(component.props().momentJsInstance.format("M")).toEqual("7");
	});
});
