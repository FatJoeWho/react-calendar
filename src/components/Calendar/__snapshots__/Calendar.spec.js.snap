// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Should render 1`] = `
<div>
  <header>
    <button
      onClick={[Function]}
    >
      ←
    </button>
    <div>
      August 2017
    </div>
    <button
      onClick={[Function]}
    >
      →
    </button>
  </header>
  <div
    className="sc-bwzfXH dZiufd"
  >
    <div>
      Sun
    </div>
    <div>
      Mon
    </div>
    <div>
      Tues
    </div>
    <div>
      Wed
    </div>
    <div>
      Thurs
    </div>
    <div>
      Fri
    </div>
    <div>
      Sat
    </div>
  </div>
  <div
    className="sc-htpNat gEZWXX"
  >
    <div
      className="sc-bdVaJa fNcbOa"
    >
      1
    </div>
    <div
      className="sc-bdVaJa fpznfK"
    >
      2
    </div>
    <div
      className="sc-bdVaJa hELaXa"
    >
      3
    </div>
    <div
      className="sc-bdVaJa bHSUef"
    >
      4
    </div>
    <div
      className="sc-bdVaJa dEOBQp"
    >
      5
    </div>
    <div
      className="sc-bdVaJa fGkvKS"
    >
      6
    </div>
    <div
      className="sc-bdVaJa idLRsj"
    >
      7
    </div>
    <div
      className="sc-bdVaJa fNcbOa"
    >
      8
    </div>
    <div
      className="sc-bdVaJa fpznfK"
    >
      9
    </div>
    <div
      className="sc-bdVaJa hELaXa"
    >
      10
    </div>
    <div
      className="sc-bdVaJa bHSUef"
    >
      11
    </div>
    <div
      className="sc-bdVaJa dEOBQp"
    >
      12
    </div>
    <div
      className="sc-bdVaJa fGkvKS"
    >
      13
    </div>
    <div
      className="sc-bdVaJa idLRsj"
    >
      14
    </div>
    <div
      className="sc-bdVaJa fNcbOa"
    >
      15
    </div>
    <div
      className="sc-bdVaJa fpznfK"
    >
      16
    </div>
    <div
      className="sc-bdVaJa hELaXa"
    >
      17
    </div>
    <div
      className="sc-bdVaJa bHSUef"
    >
      18
    </div>
    <div
      className="sc-bdVaJa dEOBQp"
    >
      19
    </div>
    <div
      className="sc-bdVaJa fGkvKS"
    >
      20
    </div>
    <div
      className="sc-bdVaJa idLRsj"
    >
      21
    </div>
    <div
      className="sc-bdVaJa fNcbOa"
    >
      22
    </div>
    <div
      className="sc-bdVaJa fpznfK"
    >
      23
    </div>
    <div
      className="sc-bdVaJa hELaXa"
    >
      24
    </div>
    <div
      className="sc-bdVaJa bHSUef"
    >
      25
    </div>
    <div
      className="sc-bdVaJa dEOBQp"
    >
      26
    </div>
    <div
      className="sc-bdVaJa fGkvKS"
    >
      27
    </div>
    <div
      className="sc-bdVaJa idLRsj"
    >
      28
    </div>
    <div
      className="sc-bdVaJa fNcbOa"
    >
      29
    </div>
    <div
      className="sc-bdVaJa fpznfK"
    >
      30
    </div>
    <div
      className="sc-bdVaJa hELaXa"
    >
      31
    </div>
  </div>
</div>
`;
